server/actions/MongoDriver.ts

```diff
@@ -13,10 +21,19 @@ const ENTITY_NOT_FOUND_MESSAGE = 'Entity does not exist'
  */
 export async function getEntity<Schema extends Document>(
   dbSchema: Model<Schema>,
-  id: string
-) {
+  id: string,
+  aggregate?: PipelineStage[]
+): Promise<Schema & { _id: ObjectId }> {
   await mongoDb()
 
+  // if aggregate is defined, use that instead of simple find
+  if (!!aggregate) {
+    const objectId = new Types.ObjectId(id)
+    aggregate.push({ $match: { _id: objectId } })
+    const response = await dbSchema.aggregate(aggregate)
+    if (!response) throw new ApiError(404, ENTITY_NOT_FOUND_MESSAGE)
+  }
+
   const response = await dbSchema.findById(id)
   if (!response) throw new ApiError(404, ENTITY_NOT_FOUND_MESSAGE)
```

Comment:
Can probably put this second block into an `else`. Only define `response` once, and share the api error and return statements.